File: Api/src/main/java/net/binggl/mydms/features/security/models/Role.java
package net.binggl.mydms.features.security.models;

/**
 * @see http://stackoverflow.com/questions/604424/lookup-enum-by-string-value
 */
public enum Role {
	None("none"), User("user"), Admin("admin");

	private String text;

	Role(String text) {
		this.text = text;
	}

	public String getText() {
		return this.text;
	}

	public static Role fromString(String text) {
		try {
			if (text != null && !"".equals(text)) {
				for (Role b : Role.values()) {
					if (text.equalsIgnoreCase(b.text)) {
						return b;
					}
				}
			}
		} catch(Exception EX) {
			
		}
		return Role.None;
	}
}

File: Api/src/main/kotlin/net/binggl/mydms/shared/models/Role.kt
package net.binggl.mydms.shared.models

enum class Role(val precedence: Int) {
    None(-1),
    User(1),
    Admin(99);

    companion object {
        fun fromString(text: String?): Role {
            if (text != null && "" != text) {
                Role.values()
                        .filter { text.equals(it.name, ignoreCase = true) }
                        .forEach { return it }
            }
            return Role.None
        }
    }
}